### PR TITLE
Add agent listing

### PR DESCRIPTION
**Pull request by cazterk**
- [x] Add agent listing
#13 

agentlisting

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+NEXT_PUBLIC_API_URI=http://localhost:8000/api/
```
Comment (sonlinux) [inline .env:1]: have  you pulled latest  changes from the `itez-platform`
`git submodule update --remote`
??

this has to read `ITEZ_API_URI`

File: src/components/Charts/ DoughnutChart.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Doughnut } from 'react-chartjs-2';
```
Comment (sonlinux) [inline src/components/Charts/ DoughnutChart.tsx:1]: this  file is names with a space at the beginning
lets not do that please
fix and  remove the space in it

File: src/components/Charts/BarChart.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  // //         barThickness: 2, // number (pixels) or 'flex'
+  // //         maxBarThickness: 1, // number (pixels)
+  // //       },
+  // //     ],
+  // //   },
+  // // };
+
+  // // const options: any = {
+  // //   cornerRadius: 4,
+  // // };
```
Comment (sonlinux) [inline src/components/Charts/BarChart.tsx:49]: if we are not using this code lets remove it from here entirely

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: src/pages/agents/index.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                          </td>
+                          <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
+                            <button className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 mt-4 mr-8 rounded">
+                              edit
+                            </button>
+                          </td>
+                        </tr>
+                      ))
+                    ) : (
+                      <p>ERROR!!!!</p>
```
Comment (sonlinux) [inline src/pages/agents/index.tsx:116]: lets handle errors properly maybe with  a  loading spinner or something not like this

File: .env (hunk 1 above)
Comment (cazterk) [reply .env]: - [x] separated the tickets

File: src/components/Charts/BarChart.tsx (hunk 3 above)
Comment (cazterk) [reply src/components/Charts/BarChart.tsx]: - [x] separated the tickets

File: src/components/Charts/ DoughnutChart.tsx (hunk 2 above)
Comment (cazterk) [reply src/components/Charts/ DoughnutChart.tsx]: - [x] separated the tickets as well

File: src/pages/agents/index.tsx (hunk 4 above)
Comment (cazterk) [reply src/pages/agents/index.tsx]: - [x] added spinner and error message

px

File: .env [hunk 5]
```diff
@@ -0,0 +1 @@
+ITEZ_API_URI=http://localhost:8000/api/
```
Comment (sonlinux) [inline .env:1]: Why is this an addition when it already existed ?

File: package.json [hunk 6]
```diff
@@ -26,6 +26,7 @@
     "next": "^12.0.2",
     "next-seo": "^4.28.1",
     "react": "^17.0.2",
+    "react-chartjs-2": "^3.3.0",
```
Comment (sonlinux) [inline package.json:29]: Is this still having charts 
I thought you said this is now segmented from the graphs work?

File: src/components/Spinner.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+const Spinner = () => {
```
Comment (sonlinux) [inline src/components/Spinner.tsx:1]: Are you rewriting and duplicating this?
Why not reuse it from the same place ?

Remove all the svgs and images we don't need 
I see additional that aren't necessary for this PR

File: src/pages/header.tsx [hunk 8]
```diff
@@ -0,0 +1,7 @@
+const Header = () => (
+  <div className="absolute z-10">
```
Comment (sonlinux) [inline src/pages/header.tsx:2]: Ma really wondering a lot why you are redoing most things when they already existed 
This is not good practice

File: src/types/beneficiary.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
+export interface Props {
```
Comment (sonlinux) [inline src/types/beneficiary.ts:1]: Everything in here was already added but it now looks like you did add them as new items 

This PR needs cleaning

Sync with upstream and make sure this Par ONLY has things relating to the table you are putting up and nothing else

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: src/types/beneficiary.ts (hunk 9 above)
Comment (cazterk) [reply src/types/beneficiary.ts:1]: - [x] removed item

File: package.json (hunk 6 above)
Comment (cazterk) [reply package.json:29]: - [x] remove package

File: src/components/Spinner.tsx (hunk 7 above)
Comment (cazterk) [reply src/components/Spinner.tsx:1]: - its a reusable component 
- [x] remove unnecessary images

File: .env (hunk 5 above)
Comment (cazterk) [reply .env:1]: - When I pulled latest changes from upstream had a conflict and accepted incoming changes I don't know if it treated it as an addition

File: src/pages/header.tsx (hunk 8 above)
Comment (cazterk) [reply src/pages/header.tsx:2]: - I have conflicts when pulling from upstream and had to accept incoming changes

**Pull request closed without merging**